tests/bluetooth/tester/src/btp_has.c

```diff
@@ -0,0 +1,208 @@
+/* btp_has.c - Bluetooth HAS Tester */
+
+/*
+ * Copyright (c) 2023 Oticon
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+#include <zephyr/bluetooth/audio/has.h>
+
+#include "btp/btp.h"
+#include "zephyr/sys/byteorder.h"
+#include "zephyr/arch/common/ffs.h"
+#include <stdint.h>
+
+#include <zephyr/logging/log.h>
+#define LOG_MODULE_NAME bttester_has
+LOG_MODULE_REGISTER(LOG_MODULE_NAME, CONFIG_BTTESTER_LOG_LEVEL);
+
+static uint8_t has_supported_commands(const void *cmd, uint16_t cmd_len,
+				      void *rsp, uint16_t *rsp_len)
+{
+	struct btp_has_read_supported_commands_rp *rp = rsp;
+
+	/* octet 0 */
+	tester_set_bit(rp->data, BTP_HAS_READ_SUPPORTED_COMMANDS);
+
+	*rsp_len = sizeof(*rp) + 1;
+
+	return BTP_STATUS_SUCCESS;
+}
+
+static uint8_t has_set_active_index(const void *cmd, uint16_t cmd_len,
+				    void *rsp, uint16_t *rsp_len)
+{
+	const struct btp_has_set_active_index_cmd *cp = cmd;
+	int err = bt_has_preset_active_set(cp->index);
+
+	return (err) ? BTP_STATUS_FAILED : BTP_STATUS_SUCCESS;
+}
+
+static uint8_t has_presets;
+static char    temp_name[BT_HAS_PRESET_NAME_MAX + 1];
+
+static uint8_t has_set_preset_name(const void *cmd, uint16_t cmd_len,
+				   void *rsp, uint16_t *rsp_len)
+{
+	const struct btp_has_set_preset_name_cmd *cp = cmd;
+	int err;
+	int name_len = MIN(cp->length, BT_HAS_PRESET_NAME_MAX);
+
+	memcpy(temp_name, cp->name, name_len);
+	temp_name[name_len] = '\0';
+	err = bt_has_preset_name_change(cp->index, temp_name);
+
+	return (err) ? BTP_STATUS_FAILED : BTP_STATUS_SUCCESS;
+}
+
+static uint8_t has_remove_preset(const void *cmd, uint16_t cmd_len,
+				 void *rsp, uint16_t *rsp_len)
+{
+	const struct btp_has_remove_preset_cmd *cp = cmd;
+	int err = 0;
+
+	if (cp->index == BT_HAS_PRESET_INDEX_NONE) {
+		uint8_t index;
+
+		while (has_presets) {
+			index = find_lsb_set(has_presets);
+			if (index) {
```

Comment:
`index` will always be true here. We can consider if it looks nicer as a loop like the following. If you like it.

```c
for (uint8_t index; index = find_lsb_set(has_presets);) {
```

Reply:
Thanks. Done.